tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -184,3 +184,201 @@ def get_resource_params(self, m, r):
+    """
+    `Patches <https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers
+    /patch>`_ configuration parameters for an instance group.
+
+    The `instanceTemplate` represents an URL of the instance template that is specified for
+    this managed instance group.
+
+    The `autoHealingPolicies[].healthCheck` represents an URL for the health check that
+    signals autohealing.
+
+    The `autoHealingPolicies[]` specifies the autohealing policy for this managed instance group.
+    You can specify only one value.
+
+    The `autoHealingPolicies[].initialDelaySec` specifies the  number of seconds that the managed
+    instance group waits before it applies autohealing policies to new instances or recently
+    recreated instances.
+
+    The `updatePolicy.type` specifies the type of update process. You can specify either
+    'PROACTIVE' so that the instance group manager proactively executes actions in order to bring
+    instances to their target versions or 'OPPORTUNISTIC' so that no action is proactively executed
+    but the update will be performed as part of other actions.
+
+    The `updatePolicy.minimalAction` specifies minimal action to be taken on an instance.  You
+    can specify either 'RESTART' to restart existing instances or 'REPLACE' to delete and create
+    new instances from the target template.
+
+    The `updatePolicy.maxSurge.fixed` specifies a fixed number of VM instances. This must be a
+    positive integer.
+    The `updatePolicy.maxSurge.percent` specifies a percentage of instances between 0 to 100%,
+    inclusive. For example, specify 80 for 80%.
+
+    The `updatePolicy.maxUnavailable.fixed` specifies a fixed number of VM instances. This must be
+    a positive integer.
+
+    The `updatePolicy.maxUnavailable.percent` specifies a percentage of instances between 0 to
+    100%, inclusive. For example, specify 80 for 80%.
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: instance-group-manager-delete
+            resource: gcp.instance-group-manager
+            actions:
+              - type: set
+                instanceTemplate:
+                  "https://www.googleapis.com/compute/v1/projects/mitrop-custodian\
+                  /global/instanceTemplates/instance-template-2"
+                autoHealingPolicies:
+                  - healthCheck:
+                      "https://www.googleapis.com/compute/v1/projects/mitrop-custodian/global\
+                      /healthChecks/test-health-check"
+                    initialDelaySec: 10
+                updatePolicy:
+                  type: OPPORTUNISTIC
+                  minimalAction: REPLACE
+                  maxSurge:
+                    fixed: 4
+                  maxUnavailable:
+                    fixed: 1
+    """
+    schema = type_schema('set',
+                         **{
+                             'instanceTemplate': {'type': 'string'},
```

Comment:
i'm a little cautious on this duplication of jsonschema, the discovery docs have the relevant schema. unfortunately gcp makes those dynamicly for sdks (although not for the cli), so its harder for us to reference them directly and avoid the duplication. we should probably distribute the sdk json files as third party data, so we can use them cleanly instead of just having them in test data.

Reply:
Should I delete the schema?

Reply:
no, this field should be validated, its more a question of not duplicating whats already in the sdk, but that requires sw dev.